src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,443 @@
+            Assert.AreEqual(15, converted, "String conversion to primitive type did not succeed.");
+        }
+
+        [TestMethod]
+        public void PropertyBag_NoCache()
+        {
+            var propertyBag = new PropertyBag();
+
+            TestEnum testEnum = propertyBag.GetProperty(TestEnumOptionTwo, cacheDefault: false);
+
+            Assert.AreEqual(TestEnumOptionTwo.DefaultValue, testEnum);
+            Assert.AreEqual(propertyBag.Count, 0);
+        }
+
+        [TestMethod]
+        public void PropertyBag_TypedPropertyBagNoCache()
+        {
+            var propertyBag = new TestEnumPropertyBag();
+
+            TestEnum testEnum = propertyBag.GetProperty(TestEnumOptionTwo, cacheDefault: false);
+
+            Assert.AreEqual(TestEnumOptionTwo.DefaultValue, testEnum);
+            Assert.AreEqual(propertyBag.Count, 0);
+        }
+
+        [TestMethod]
+        public void PropertyBag_InitializeFromPropertyBag()
+        {
+            int expectedValue = 72;
+            var propertyBag = new PropertyBag();
+            propertyBag["TestValue"] = expectedValue.ToString();
+            
+            var copiedPropertyBag = new PropertyBag(propertyBag, StringComparer.OrdinalIgnoreCase);
+
+            int converted;
+            Assert.IsTrue(propertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+
+            Assert.IsFalse(propertyBag.TryGetProperty("testvalue", out converted));
+
+            Assert.IsTrue(copiedPropertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+
+            Assert.IsTrue(copiedPropertyBag.TryGetProperty("testvalue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+        }
+
+        [TestMethod]
+        public void PropertyBag_SetProperty()
+        {
+            var propertyBag = new PropertyBag();
+
+            var descriptor = new PerLanguageOption<string>("feature", "name", "defaultValue");
+
+            propertyBag.SetProperty(descriptor, "value");
+            Assert.IsTrue(propertyBag.GetProperty(descriptor) == "value");
+
+            propertyBag.SetProperty(descriptor, null);
+            Assert.IsTrue(propertyBag.GetProperty(descriptor) == "defaultValue");
+        }
+
+        [TestMethod]
+        public void PropertyBag_BinarySerialization()
+        {
+            var testData = new TestData();
+            var propertyBag = new PropertyBag();
+            testData.InitializePropertyBag(propertyBag);
+
+
+            var formatter = new BinaryFormatter();
+
+            string path = Path.GetTempFileName();
+
+            try
+            {
+                using (var writer = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite))
```

Comment:
Could you use a `MemoryStream` instead of `FileStream` and avoid having to get a temp file name, and deleting a file, and catching `IOException`? It would run faster too.